api/app/models/spree/api/webhooks/stock_item_decorator.rb

```diff
@@ -0,0 +1,22 @@
+module Spree
+  module Api
+    module Webhooks
+      module StockItemDecorator
+        def self.prepended(base)
+          base.after_commit :queue_webhooks_requests_for_sold_out_products
+        end
+
+        private
+
+        def queue_webhooks_requests_for_sold_out_products
+          return if deleted_at.nil? && !count_on_hand_previously_changed?
+          return if product.in_stock?
```

Comment:
@KacperMekarski I'm afraid this could be wrong because of caching / memoizations as this method is both cached and memorized, so maybe running an SQL query and checking count_on_hand and backorderable in stock items would be better

Reply:
@aleksandarpetrushev maybe you could take over this one? probably this fix will be required for all places checking `in_stock?` when Stock Items change